### DISCUSSION TITLE
Query Parameter Options ERROR

### DISCUSSION THREAD

**Discussion by matjazreberc** [microsoftgraph/msgraph-sdk-dotnet#2119] [Q&A] [answered]
Hello!

I want to get SP List items, but i get error: "Exception of type 'Microsoft.Graph.Models.ODataErrors.ODataError' was thrown."

My code:

`
                
                var items = await graphClient.Sites[siteCollection.Id].Lists[listId].Items.GetAsync((requestConfiguration) =>
                    {
                        requestConfiguration.QueryParameters.Select = new string[] {"ID","iPM_ID"};
                        //requestConfiguration.QueryParameters.Expand = new string[] { "Requestor" };

                    });`

If i only want to return "ID" then it works, but if I want to return more data i get error.
Why?

In older version of MS Graph 4.xxx  i can get all data that i want.

Thank you, Matjaz Reberc

**Comment by andrueastman** [comment]
Any chance you can share the error (code and message) details by following the guidance at the link below? 
https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/errors.md#handling-errors-in-the-microsoft-graph-net-client-library

**Reply by matjazreberc** [reply]
Hello,

I addet "catch (ODataError odataError)"

This is the error message:  "Parsing OData Select and Expand failed: Could not find a property named 'iPM_ID' on type 'microsoft.graph.listItem'."

The next thing i did was to get all columns from List, 

`var result = await graphClient.Sites[siteCollection.Id].Lists[listId].Columns.GetAsync();`

and "iPM_ID"  is in this List

![image](https://github.com/microsoftgraph/msgraph-sdk-dotnet/assets/60425382/7d87a21b-f023-45de-ba73-66f8f94e0b3a)


The Sitecollection id  and List id are the same.
How can this be?

Thank you, Matjaz

**Reply by andrueastman** [reply]
Can you by any chance share the code that works on V4 of the SDK? V5 adds query parameters as `$select` in the url. Are you by any change using `select`?

**Reply by matjazreberc** [reply]
Hi,

this is my V4 code, (i delete part of the code):

`var queryOptions = new List<QueryOption>()
                {
                    new QueryOption("expand", "fields(select=ID,iPM_ID,...)")
                };`
`

                var itemsList = new List<ListItem>();
                var items = await graphClient.Sites[siteCollection.Id].Lists[listId].Items
                    .Request(queryOptions)
                    .Filter("(Fields/Phase eq ...')")
                    .GetAsync();
`

and new code

` var items = await graphClient.Sites[siteCollection.Id].Lists[listId].Items.GetAsync(requestConfiguration =>{
                        //requestConfiguration.QueryParameters.Select = new string[] { "ID" };

                        //requestConfiguration.QueryParameters.Select = new string[] {"ID,Title"};

                        requestConfiguration.QueryParameters.Select = new string[] { "ID","Title" };

                        //requestConfiguration.QueryParameters.Expand = new string[] { "Requestor" };
                    });
`

I try to use this documentation: 
https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/upgrade-to-v5.md

https://learn.microsoft.com/en-us/graph/query-parameters?tabs=csharp

https://learn.microsoft.com/en-us/graph/api/listitem-list?view=graph-rest-1.0&tabs=csharp

Thank, Matjaz

**Reply by andrueastman** [reply, marked as answer]
The select looks to be nested in the expand and should not be on the top level. 

Any chance, the following code works for you?
```cs
var result = await graphClient.Sites["{site-id}"].Lists["{list-id}"].Items.GetAsync((requestConfiguration) =>
{
	requestConfiguration.QueryParameters.Expand = new string []{ "fields($select=id,Edit)" };
});
```

**Reply by matjazreberc** [reply]
It works.

My code (before)
`requestConfiguration.QueryParameters.Expand = new string[] { "fields(select=ID,Title)" };`

Your code:
`requestConfiguration.QueryParameters.Expand = new string []{ "fields($select=id,Edit)" };`

Difference: **$**  before "select=..."

Didn't see this in documentation.

Thank you, Matjaz

**Comment by matjazreberc** [comment]
Problem was missing  **$** before select.

Matjaz
